backend/utils/bulk_import.py

```
@@ -0,0 +1,448 @@
+        'requires_calibration': str(row_data.get('requires_calibration', 'false')).lower() in ['true', '1', 'yes'],
+        'calibration_frequency_days': None
+    }
+    
+    # Handle calibration frequency
+    if tool_data['requires_calibration']:
+        freq_str = row_data.get('calibration_frequency_days', '')
+        if freq_str:
+            try:
+                tool_data['calibration_frequency_days'] = int(freq_str)
+            except ValueError:
+                raise ValidationError(f"Invalid calibration frequency: {freq_str}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)